include/engine/api/match_parameters.hpp

```diff
@@ -61,12 +62,16 @@ struct MatchParameters : public RouteParameters
     }
 
     template <typename... Args>
-    MatchParameters(std::vector<unsigned> timestamps_, Args... args_)
-        : RouteParameters{std::forward<Args>(args_)...}, timestamps{std::move(timestamps_)}
+    MatchParameters(
+        std::vector<unsigned> timestamps_,
+        const boost::optional<bool> allow_uturn_,
+        Args... args_)
+        : RouteParameters{std::forward<Args>(args_)...}, timestamps{std::move(timestamps_), allow_uturn{allow_uturn_}}
```

Comment:
This needs to be done in a backwards compatible way: this is libosrm's public interface which we can't change without bumping the major version number.

You could add a constructor overload taking the new parameter and keeping the old one as is. Then add a note (and once we merged this PR open a ticket) for removing this backwards compatible workaround in v6. And tag it with the milestone v6.